### PR TITLE
ISICs segmentation with U-Net (Declan Healey)

### PR DESCRIPTION
**Pull request by dhealey95**
Hi Shakes,

Please review this pull request containing an implementation for solving the ISICs segmentation problem using the U-Net. When trained to 30 epochs, the model achieved a Dice coefficient of over 0.7 for predicting the segmentation mask in the original images in the dataset.

- **README.md** - contains the relevant documentation for the solution implemented.
- **layers.py** - defines the layers in the U-Net CNN.
- **test_unet_isics.py** - driver script for pre-processing the dataset, building the model and retrieving results.

Thanks,

Declan Healey
Student No: 43560734

### CODE REVIEW

**Comment by SiyuLiu0329** [comment]
- Work submitted before deadline 
- Code appear valid for Easy Difficulty 
- External sources appropriately credited 
- Results appear good 
- Sufficiently details commit messages
Other Feedback
- Good work. I would also u-net-architecture.png in a sub folder

**Comment by rexsw** [comment]
Code looks good for easy difficulty

For next time;

Make sure you remove unused imports

Tensorflow has functions for train-test splitting and more so there’s no need to also use sklearn in your project. 

Paths such as ‘C:\\data\ISIC2018_Task1-2_Training_Data\ISIC2018_Task1-2_Training_Input_x2\*.jpg’ won’t work for the user and should either be set dynamical or it needs to be made clear to the user that they need to change these paths

Comments such as 
def display(display_list):
    """
    Displays a list of images.
Add nothing to the readability of the code and should be avoided. Overall comments were well done

**Comment by shakes76** [comment]
## TF Usage
OK
## Good Practice (Design/Commenting)
Good design and comments
## Algorithm 
Driver Script present
Shows Usage & Demo & Visualisation & Data usage, Readme OK
Module present
## Commit Log
Meaningful commit messages 
Progressive commits used 
## Documentation
Description and Comments good
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
In the future, a more detailed pull request could've been done. When developer receiving the request is not known, the request could be rejected. Remember you could use/reuse the README info.
